ls1intum/Artemis
# `Text exercises`: Hide feedback section when tutor has assessment lock after deadline

### Checklist
#### General


#### Client

### Motivation and Context


Fixes #1575 

### Description

I added an explicit check for the case where a result exists, but no feedback (lock). I additionally refactored the section a bit.

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Student
- 1 Text exercise with student submission and due date and assessment due date passed

- View the submission as a student: You should be able to submit late
- Acquire a lock for the student submission as tutor
- View the submission as a student: You should not be able to submit late anymore, nothing should indicate that your submission got already assessed
- Remove the lock as tutor
- View the submission as student: You should be able to submit late again

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots


Before:
![](https://gyazo.com/832a436d38a7758afa839177a5a71487/raw)
Now:
![](https://gyazo.com/12f8a748ae9a7b7f20e9e5f759eaaadf/raw)
